src/database.c

```diff
@@ -0,0 +1,179 @@
+/*
+ * Punica - LwM2M server with REST API
+ * Copyright (C) 2018 8devices
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "settings.h"
+#include "rest-core-types.h"
+#include "rest-list.h"
+#include "restserver.h"
+
+#define DATABASE_UUID_KEY_BIT       0x1
+#define DATABASE_PSK_KEY_BIT        0x2
+#define DATABASE_PSK_ID_KEY_BIT     0x4
+#define DATABASE_ALL_KEYS_SET       0x7
+
+int database_load_file(rest_context_t *rest)
+{
+    json_error_t error;
+    const char *section;
+    size_t index;
+    json_t *j_value;
+    json_t *j_entry;
+    json_t *j_database = NULL;
+    int key_check;
+    int ret = 1;
```

Comment:
although using `ret` `curr` and similar variable names is syntactically correct, and purpose is evident, but I would like to see more verbose variable names, because it doesn't increase typing time or code complexity by much, but drastically improves readability (personal opinion again).